### ISSUE TITLE
Thoughts on adding a facility to enable users to send technical & diagnostic info from the app?

### ISSUE THREAD

**Issue by julianharty** [kiwix/kiwix-android#305]
Sometimes our app fails our users. Some of them may be willing to provide information that'll help us to help them diagnose possible causes. While they can use Android's crash reporting and/or write a review on the Google Playstore, we're unlikely to get specific information from either of these methods, and few will venture to github or use http://bugs.kiwix.org

One way we could help users provide technical information is by adding a facility to the app that collects relevant technical information e.g. on the device details, the location of file storage, etc. I propose they'd be in control of the process and sending the information.

Here are some suggestions of the transmission methods:

- email (e.g. using an Android Intent) where we format the email with contents (perhaps as a structured message that's easy and reliable to process).
- a HTTP post to a server we host
- a third-party bug/analytics reporting service
We can discuss the options, concerns, etc. here and/or elsewhere rather than me try to frame and add lots of detail here.

See #304 for some previous related notes. We could consider adding the reporting facility as a generally available function (e.g. on the settings menu) and/or in a global error handler such as described in that issue. We could consider adding this feature to the main app, or in diagnostic builds.

Ideas, thoughts & concerns welcome, as would be a possible design and suggestions of what data would be most relevant and useful.

PS: this isn't unique to the Android app, we might want to do something similar in other apps.

**Comment by mhutti1** [comment]
I think this would help a lot but don't really have any idea how to start out with an implementation.

**Comment by julianharty** [comment]
Very good, I'll have a go at creating something skeletal so we can try it out in the app. Here's a suggestion https://stackoverflow.com/a/19968400/340175 I've also created something many years ago that was pretty basic (I'd like the one we provide in Kiwix-Android to collect data I didn't need in that app). I'll ping you with the branch with the skeletal code, however as-ever you're welcome to work in parallel.

**Comment by julianharty** [comment]
And in that vein, I'd like to capture the information from some of your (@mhutti1) code.

**Comment by kelson42** [comment]
Yes, but please do not use email because this is neither transparent nor reliable. I vote for an online plaftorm and the usage of a standard solution... Otherwise maybe we could just achieve easily to things posted as bug on Github. As long as we do not distribute intensively we could deal with the stream of reports IMO.

**Comment by kelson42** [comment]
@julianharty I think this is now pretty much a duplicate of #335!? Should we close that ticket?

**Comment by julianharty** [comment]
Closing as we've several newer tickets on this topic, follow #335 if you're interested in the chain of tickets.
